pkg/operator/status/controller.go

```diff
@@ -100,13 +101,36 @@ func (c StatusSyncer) sync() error {
 	if availableCondition != nil {
 		conditions = append(conditions, availableCondition)
 	}
-	failingCondition, err := OperatorConditionToClusterOperatorCondition(v1helpers.FindOperatorCondition(currentDetailedStatus.Conditions, operatorv1.OperatorStatusTypeFailing))
-	if err != nil {
-		return err
+
+	var failingConditions []operatorv1.OperatorCondition
+	for _, condition := range currentDetailedStatus.Conditions {
+		if strings.HasSuffix(condition.Type, "Failing") && condition.Status == operatorv1.ConditionTrue {
+			failingConditions = append(failingConditions, condition)
+		}
 	}
+	var failingCondition map[string]interface{}
+	if len(failingConditions) > 0 {
+		failingCondition = map[string]interface{}{}
+		unstructured.SetNestedField(failingCondition, operatorv1.OperatorStatusTypeFailing, "Type")
+		unstructured.SetNestedField(failingCondition, string(operatorv1.ConditionTrue), "Status")
+		var messages []string
+		for _, condition := range failingConditions {
+			if len(condition.Message) == 0 {
+				continue
+			}
+			for _, message := range strings.Split(condition.Message, "\n") {
+				messages = append(messages, fmt.Sprintf("%s: %s", condition.Type, message))
+			}
+		}
+		if len(messages) > 0 {
+			unstructured.SetNestedField(failingCondition, strings.Join(messages, "\n"), "Message")
+		}
+	}
```

Comment:
else, explicitly set the condition to false

Reply:
:+1: